api/base.ts

```diff
@@ -0,0 +1,15 @@
+import axios from "axios"
+const base_url = process.env.NEXT_PUBLIC_BASE_URL
+
+export const base = axios.create({
+  baseURL: base_url,
+  headers: { "Content-Type": "application/json" },
+})
+
+export const baseWithAuth = () => {
+  const token = localStorage.getItem("userToken")
+  if (token) {
+    base.defaults.headers.common["authorization"] = JSON.parse(token)
```

Comment:
Saw it @macarron001. Why are you setting the `default header` as the JSON.parse(token)? It shouldn't be the default. 

The token is read from the localStorage. So in effect, this is caching the header and will be sent `for all requests` moving forward.

Reply:
ohh I see

Reply:
I thought it was only changing a specific instance of that request

Reply:
I'll just put it back to how it was previously. Axios doesn't allow pala kasi to extend from an existing instance unlike the way it could be done sa wretch that I saw sa tempalate.

Reply:
good catch sir! thanks 😁